### PR TITLE
E2e date transform

### PR DESCRIPTION
**Pull request by Swati10-11**
@sawantpritam Please Review

### CODE REVIEW

File: e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature [hunk 1]
```diff
@@ -0,0 +1,159 @@
+@DateTransform
```
Comment (sawantpritam) [inline e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature]: rename file -> DateTransformWithBQ.featrure

File: e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+Feature:Verification of DateTransform Plugin
```
Comment (sawantpritam) [inline e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature]: -> Feature: DateTransform - Verification of DateTransform pipeline with BigQuery as source and target

File: e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Close the pipeline logs
+    Then Validate OUT record count is equal to IN record count
+
+  @BQ_SINK_TEST @BQ_SOURCE_DATETRANSFORM_LONG_TEST @PLUGIN-1225
+  Scenario: To verify complete flow of data extract and transfer from BQ source to BQ sink using DateTransform Plugin with Long field
```
Comment (sawantpritam) [inline e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature]: -> using DateTransform Plugin for long datatype field

File: e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature [hunk 4]
```diff
@@ -0,0 +1,159 @@
+@DateTransform
+Feature:Verification of DateTransform Plugin
+
+  @BQ_SINK_TEST @BQ_SOURCE_DATETRANSFORM_TEST
+  Scenario: To verify complete flow of data extract and transfer from BQ source to BQ sink using DateTransform Plugin
```
Comment (sawantpritam) [inline e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature]: -> using DateTransform Plugin for string datatype field

File: e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    Then Click on the Preview Data link on the Sink plugin node: "BigQueryTable"
+    Then Verify sink plugin's Preview Data for Input Records table and the Input Schema matches the Output Schema of Source plugin
+    Then Close the preview data
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Close the pipeline logs
+    Then Validate OUT record count is equal to IN record count
```
Comment (sawantpritam) [inline e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature]: Add format validation step

Applies to all scenarios

File: e2e-test/src/test/features/datetransformplugin/datetransformErrorScenarios.feature [hunk 6]
```diff
@@ -0,0 +1,105 @@
+@DateTransform
```
Comment (sawantpritam) [inline e2e-test/src/test/features/datetransformplugin/datetransformErrorScenarios.feature]: nit filename -> DateTransformErrorScenarios.feature

File: e2e-test/src/test/features/datetransformplugin/datetransformErrorScenarios.feature [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    Then Validate "BigQuery" plugin properties
+    Then Close the Plugin Properties page
+    Then Navigate to the properties page of plugin: "DateTransform"
+    Then Enter input plugin property: "SourceFieldName" with value: "SourceFieldName"
+    Then Enter input plugin property: "TargetFieldName" with value: "targetfieldNameError"
+    Then Click on the Validate button
+    Then Verify that the Plugin is displaying an error message: "errorMessageDateTransformMustHaveSameNumberOfFields" on the header
+
+  @BQ_SOURCE_DATETRANSFORM_TEST
+  Scenario:Verify DateTransform plugin error for Source and Target field must have same number of fields
```
Comment (sawantpritam) [inline e2e-test/src/test/features/datetransformplugin/datetransformErrorScenarios.feature]: Scenario description does not match with validation

File: e2e-test/src/test/features/datetransformplugin/datetransformMacros.feature [hunk 8]
```diff
@@ -0,0 +1,64 @@
+@DateTransform
+Feature:Verification of DateTransform Plugin with macros
```
Comment (sawantpritam) [inline e2e-test/src/test/features/datetransformplugin/datetransformMacros.feature]: -> Feature: DateTransform - Verification of DateTransform pipeline with BigQuery as source and target using macros

File: e2e-test/src/test/features/datetransformplugin/datetransformMacros.feature [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+  @BQ_SINK_TEST @BQ_SOURCE_DATETRANSFORM_TEST @PLUGIN-1224
+  Scenario: To verify data is getting transferred from BigQuery to BigQuery successfully with DateTransform as macro arguments
```
Comment (sawantpritam) [inline e2e-test/src/test/features/datetransformplugin/datetransformMacros.feature]: with DateTransform plugin properties as macro arguments

File: e2e-test/src/test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 10]
```diff
@@ -16,26 +16,35 @@
...
  * GCP test hooks.
  */
 public class TestSetupHooks {
+  public static String gcsSourceBucketName = StringUtils.EMPTY;
+  public static String gcsTargetBucketName = StringUtils.EMPTY;
   private static boolean firstFileSourceTestFlag = true;
   private static boolean firstFileSinkTestFlag = true;
 
-  @Before(order = 1, value = "@FILE_SOURCE_TEST")
+  @Before(order = 1, value = "@FILE_SOURCE_TEST ")
```
Comment (sawantpritam) [inline e2e-test/src/test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: Unnecessary change - space added 

-> @Before(order = 1, value = "@FILE_SOURCE_TEST")

File: e2e-test/src/test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 11]
```diff
@@ -117,11 +126,99 @@ public static void createTempSourceBQTable() throws IOException, InterruptedExce
...
     PluginPropertyUtils.removePluginProp("bqSourceTable");
   }
+
+  @Before(order = 1, value = "@BQ_SOURCE_DATETRANSFORM_TEST")       // changes for DateTransform
+  public static void createSourceBQTableWithDifferentDataTypes() throws IOException, InterruptedException {
+    createSourceBQTableWithQueries(PluginPropertyUtils.pluginProp("dateTransformBQCreateTableQueryFile"),
+                                   PluginPropertyUtils.pluginProp("dateTransformBQInsertDataQueryFile"));
+  }
+
+  @Before(order = 1, value = "@BQ_SOURCE_DATETRANSFORM_LONG_TEST")       // changes for DateTransform
```
Comment (sawantpritam) [inline e2e-test/src/test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: remove @BQ_SOURCE_DATETRANSFORM_LONG_TEST hook

File: e2e-test/src/test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 12]
```diff
@@ -117,11 +126,99 @@ public static void createTempSourceBQTable() throws IOException, InterruptedExce
...
+      BigQueryClient.getSoleQueryResult(insertDataQuery);
+    } catch (NoSuchElementException e) {
+      // Insert query does not return any record.
+      // Iterator on TableResult values in getSoleQueryResult method throws NoSuchElementException
+    }
+    PluginPropertyUtils.addPluginProp("bqSourceTable", bqSourceTable);
+    BeforeActions.scenario.write("BQ Source Table " + bqSourceTable + " created successfully");
+  }
+
+  @Before(order = 1, value = "@GCS_SOURCE_DATETRANSFORM_TEST")
```
Comment (sawantpritam) [inline e2e-test/src/test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: Remove all GCS hooks related code, as you are not using it.

File: e2e-test/src/test/java/io/cdap/plugin/datetransform/runners/TestRunner.java [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * Test Runner to execute DateTransform plugin testcases.
+ */
+@RunWith(Cucumber.class)
+@CucumberOptions(
+  features = {"src/test/features"},
+  glue = {"stepsdesign", "io.cdap.plugin.common.stepsdesign"},
+  tags = {"@DateTransform and not @PLUGIN-1224 and not @PLUGIN-1225"},
+  //TODO: Enable test once issue is fixed https://cdap.atlassian.net/browse/PLUGIN-1224
```
Comment (sawantpritam) [inline e2e-test/src/test/java/io/cdap/plugin/datetransform/runners/TestRunner.java]: `/* TODO: Enable test once issues fixed - https://cdap.atlassian.net/browse/PLUGIN-1224 https://cdap.atlassian.net/browse/PLUGIN-1225 */`

File: e2e-test/src/test/java/io/cdap/plugin/datetransform/runners/package-info.java [hunk 14]
```diff
@@ -0,0 +1 @@
+package io.cdap.plugin.datetransform.runners;
```
Comment (sawantpritam) [inline e2e-test/src/test/java/io/cdap/plugin/datetransform/runners/package-info.java]: add javadoc

File: e2e-test/src/test/java/io/cdap/plugin/utils/E2ETestConstants.java [hunk 15]
```diff
@@ -20,4 +20,13 @@
  */
 public class E2ETestConstants {
   public static final String ERROR_MSG_FILE_INVALID_OUTPUTFIELD = "errorMessageFileInvalidOutputField";
+  public static final String ERROR_MSG_DATETRANSFORM_INVALID_SOURCE_FIELD_NAME = "errorMessageDateTransformInvalid" +
```
Comment (sawantpritam) [inline e2e-test/src/test/java/io/cdap/plugin/utils/E2ETestConstants.java]: No need to add constants here as your are using error messages in scenarios directly from errorMessage.properties file

File: e2e-test/src/test/resources/errorMessage.properties [hunk 16]
```diff
@@ -2,3 +2,11 @@ validationSuccessMessage=No errors found.
 validationErrorMessage=COUNT ERROR found
 errorMessageInputPath=Invalid schema: Input path not found
 errorMessageFileInvalidOutputField=Path field 'PATH_FIELD' must exist in input schema.
+errorMessageDateTransformInvalidSourceFieldName=Null error occurred while configuring the stage DateTransform.
+errorMessageDateTransformInvalidTargetFieldName=Error encountered while configuring the stage: 'Target Field must exist \
+  in output schema.'
+errorMessageForBlankSourceField=Required property 'sourceField' has no value.
```
Comment (sawantpritam) [inline e2e-test/src/test/resources/errorMessage.properties]: remove all unused - > Required property error message

File: e2e-test/src/test/resources/pluginDataCyAttributes.properties [hunk 17]
```diff
@@ -24,3 +24,12 @@ readFilesRecursively=recursive
 allowEmptyInput=ignoreNonExistingFolders
 fileSystemProperties=fileSystemProperties
 fileEncoding=fileEncoding
+SourceFieldName=sourceField
```
Comment (sawantpritam) [inline e2e-test/src/test/resources/pluginDataCyAttributes.properties]: use camelcase for all 
i.e. SourceFieldName -> sourceFieldName

File: e2e-test/src/test/resources/pluginParameters.properties [hunk 18]
```diff
@@ -48,3 +48,23 @@ filePluginOutputFolder=file-plugin-output
...
 ## FILE-PLUGIN-PROPERTIES-END
+
+## DateTransform-PLUGIN-PROPERTIES-START
+dateTransformBQCreateTableQueryFile=testdata/BigQuery/bqDateTransformCreateTableQuery.txt
+dateTransformBQInsertDataQueryFile=testdata/BigQuery/bqDateTransformInsertQuery.txt
+dateTransformBQCreateTableQueryFile1=testdata/BigQuery/bqDateTransformCreateTableLong.txt
+dateTransformBQInsertDataQueryFile1=testdata/BigQuery/bqDateTransformInsertQueryLong.txt
+dateTransformGcsCsvFile=testdata/DateTransformGCS/gcsDateTransform.csv
+dateTransformGcsCsvFileSchema=[{"key":"id","value":"int"},{"key":"name","value":"string"},{"key":"date_of_birth","value":"int"}
+sourceFieldName=date_of_admission
```
Comment (sawantpritam) [inline e2e-test/src/test/resources/pluginParameters.properties]: add 'dateTransform' prefix for all properties 
i.e sourceFieldName -> dateTransformSourceFieldName or dateTransform.sourceFieldName

File: e2e-test/src/test/resources/pluginParameters.properties [hunk 19]
```diff
@@ -48,3 +48,23 @@ filePluginOutputFolder=file-plugin-output
...
+targetFieldName=date_of_admission
+sourceFieldDateFormat=yyyy-mm-dd
+targetFieldDateFormat=dd-mm-yyyy
+SourceFieldName=transaction_date,date_of_admission,date_of_joining
```
Comment (sawantpritam) [inline e2e-test/src/test/resources/pluginParameters.properties]: Names

File: e2e-test/src/test/resources/pluginParameters.properties [hunk 20]
```diff
@@ -48,3 +48,23 @@ filePluginOutputFolder=file-plugin-output
...
+TargetFieldName=transaction_date,date_of_admission,date_of_joining
+SourceFieldDateFormat=yyyy-mm-dd
```
Comment (sawantpritam) [inline e2e-test/src/test/resources/pluginParameters.properties]: duplicate

File: e2e-test/src/test/resources/pluginParameters.properties [hunk 21]
```diff
@@ -48,3 +48,23 @@ filePluginOutputFolder=file-plugin-output
...
+TargetFieldDateFormat=dd-mm-yyyy
+IncorrectSourceFieldName=@#$%^&
+IncorrectTargetFieldName=@#$%^&
+targetfieldNameError=transaction_date
```
Comment (sawantpritam) [inline e2e-test/src/test/resources/pluginParameters.properties]: no need of this -> instead u can use "Single field in source" and "multiple fields property in target" to get error.

File: e2e-test/src/test/resources/testdata/BigQuery/bqDateTransformCreateTableLong.txt [hunk 22]
```diff
@@ -0,0 +1 @@
+create table `DATASET.TABLE_NAME` ( Name STRING, date_of_admission INT64)
```
Comment (sawantpritam) [inline e2e-test/src/test/resources/testdata/BigQuery/bqDateTransformCreateTableLong.txt]: As discussed combine this in other table

**Review by sawantpritam — CHANGES_REQUESTED** [review]

File: e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature (hunk 1 above)
Comment (Swati10-11) [reply e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature:1]: Renamed File

File: e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature (hunk 2 above)
Comment (Swati10-11) [reply e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature:2]: @sawantpritam Changes Done

File: e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature (hunk 3 above)
Comment (Swati10-11) [reply e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature:109]: @sawantpritam changes done

File: e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature (hunk 4 above)
Comment (Swati10-11) [reply e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature:5]: @sawantpritam  changes done

File: e2e-test/src/test/features/datetransformplugin/DateTransformErrorScenarios.feature [hunk 23]
```diff
@@ -0,0 +1,105 @@
...
+    Then Enter input plugin property: "referenceName" with value: "BQReferenceName"
+    Then Enter input plugin property: "dataset" with value: "dataset"
+    Then Enter input plugin property: "table" with value: "bqSourceTable"
+    Then Click on the Get Schema button
+    Then Capture the generated Output Schema
+    Then Validate "BigQuery" plugin properties
+    Then Close the Plugin Properties page
+    Then Navigate to the properties page of plugin: "DateTransform"
+    Then Enter input plugin property: "sourceFieldName" with value: "dateTransform.SourceFieldNames"
+    Then Enter input plugin property: "targetFieldName" with value: "dateTransform.TargetFieldNameError"
```
Comment (sawantpritam) [inline e2e-test/src/test/features/datetransformplugin/DateTransformErrorScenarios.feature]: dateTransform.TargetFieldNameError -> dateTransform.TargetFieldName

File: e2e-test/src/test/resources/pluginParameters.properties [hunk 24]
```diff
@@ -0,0 +1,65 @@
...
+dateTransform.SourceFieldName=date_of_admission
+dateTransform.TargetFieldName=date_of_admission
+dateTransform.SourceFieldDateFormat=yyyy-mm-dd
+dateTransform.TargetFieldDateFormat=dd-mm-yyyy
+dateTransform.SourceFieldNames=date_of_admission,date_of_joining
+dateTransform.TargetFieldNames=date_of_admission,date_of_joining
+dateTransform.SourceFieldNameWithLongDataType=transaction_date
+dateTransform.TargetFieldNameWithLongDataType=transaction_date
+dateTransform.IncorrectFieldName=@#$%^&
+dateTransform.TargetFieldNameError=transaction_uid,date_of_joining,date_of_admission
```
Comment (sawantpritam) [inline e2e-test/src/test/resources/pluginParameters.properties]: remove this -> dateTransform.TargetFieldNameError

File: e2e-test/src/test/features/datetransformplugin/DateTransformErrorScenarios.feature [hunk 25]
```diff
@@ -0,0 +1,105 @@
...
+    Then Validate "BigQuery" plugin properties
+    Then Close the Plugin Properties page
+    Then Navigate to the properties page of plugin: "DateTransform"
+    Then Enter input plugin property: "sourceFieldName" with value: "dateTransform.SourceFieldName"
+    Then Enter input plugin property: "targetFieldName" with value: "dateTransform.IncorrectFieldName"
+    Then Click on the Validate button
+    Then Verify that the Plugin is displaying an error message: "errorMessageDateTransformInvalidTargetFieldName" on the header
+
+  @BQ_SOURCE_DATETRANSFORM_TEST
+  Scenario:Verify DateTransform plugin error for single Source and Target field must have same number of fields
```
Comment (sawantpritam) [inline e2e-test/src/test/features/datetransformplugin/DateTransformErrorScenarios.feature]: Verify DateTransform plugin error for Source and Target field must have same number of fields

File: e2e-test/src/test/features/datetransformplugin/DateTransformWithBQ.feature [hunk 26]
```diff
@@ -0,0 +1,162 @@
...
+    Then Verify sink plugin's Preview Data for Input Records table and the Input Schema matches the Output Schema of Source plugin
+    Then Close the preview data
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Close the pipeline logs
+    Then Validate OUT record count is equal to IN record count
+    Then Validate dateFormat "dateTransform.TargetFieldDateFormat" of the fields "dateTransform.TargetFieldName" in target BQ table "bqTargetTable"
```
Comment (sawantpritam) [inline e2e-test/src/test/features/datetransformplugin/DateTransformWithBQ.feature]: dateTransform.TargetFieldName -> dateTransform.TargetFieldNames

File: e2e-test/src/test/features/datetransformplugin/DateTransformWithBQ.feature [hunk 27]
```diff
@@ -0,0 +1,162 @@
...
+    Then Enter input plugin property: "dataset" with value: "dataset"
+    Then Enter input plugin property: "table" with value: "bqSourceTable"
+    Then Click on the Get Schema button
+    Then Capture the generated Output Schema
+    Then Validate "BigQuery" plugin properties
+    Then Close the Plugin Properties page
+    Then Navigate to the properties page of plugin: "DateTransform"
+    Then Enter input plugin property: "sourceFieldName" with value: "dateTransform.SourceFieldNameWithLongDataType"
+    Then Enter input plugin property: "sourceFieldDateFormat" with value: "dateTransform.SourceFieldDateFormat"
+    Then Select dropdown plugin property: "SourceinSecondsorMilliseconds?" with option value: "dateTransform.sourceinSecondsorMilliseconds?"
```
Comment (sawantpritam) [inline e2e-test/src/test/features/datetransformplugin/DateTransformWithBQ.feature]: remove ? from property name

Also value property is not added in pluginParameters.properties

File: e2e-test/src/test/features/datetransformplugin/datetransformMacros.feature [hunk 28]
```diff
@@ -0,0 +1,65 @@
+@DateTransform
+Feature:DateTransform - Verification of DateTransform pipeline with BigQuery as source and target using macros
+
+  @BQ_SINK_TEST @BQ_SOURCE_DATETRANSFORM_TEST @PLUGIN-1224
+  Scenario: To verify data is getting transferred from BigQuery to BigQuery successfully with DateTransform as macro arguments
```
Comment (sawantpritam) [inline e2e-test/src/test/features/datetransformplugin/datetransformMacros.feature]: -> with DateTransform plugin properties as macro arguments

File: e2e-test/src/test/java/io/cdap/plugin/datetransform/runners/TestRunner.java [hunk 29]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * Test Runner to execute DateTransform plugin testcases.
+ */
+@RunWith(Cucumber.class)
+@CucumberOptions(
+  features = {"src/test/features"},
+  glue = {"io.cdap.plugin.datetransform.stepsdesign", "stepsdesign", "io.cdap.plugin.common.stepsdesign"},
+  tags = {"@DateTransform and not @PLUGIN-1224 and not @PLUGIN-1225"},
+  //TODO: Enable test once issue is fixed https://cdap.atlassian.net/browse/PLUGIN-1224
```
Comment (sawantpritam) [inline e2e-test/src/test/java/io/cdap/plugin/datetransform/runners/TestRunner.java]: Add multiline comment ->

`/* TODO: Enable test once issues fixed - https://cdap.atlassian.net/browse/PLUGIN-1224 https://cdap.atlassian.net/browse/PLUGIN-1225 */`

File: e2e-test/src/test/java/io/cdap/plugin/datetransform/runners/package-info.java (hunk 14 above)
Comment (sawantpritam) [inline e2e-test/src/test/java/io/cdap/plugin/datetransform/runners/package-info.java]: Add javadoc 
and also licence.

File: e2e-test/src/test/java/io/cdap/plugin/datetransform/stepsdesign/DateTransform.java [hunk 30]
```diff
@@ -0,0 +1,60 @@
...
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Optional;
+
+/**
+ * DateTransform related Stepsdesign.
+ */
+
+public class DateTransform implements CdfHelper {
+  @Then("Validate dateFormat {string} of the fields {string} in target BQ table {string}")
```
Comment (sawantpritam) [inline e2e-test/src/test/java/io/cdap/plugin/datetransform/stepsdesign/DateTransform.java]: Add one empty line before

File: e2e-test/src/test/java/io/cdap/plugin/datetransform/stepsdesign/package-info.java [hunk 31]
```diff
@@ -0,0 +1 @@
+package io.cdap.plugin.datetransform.stepsdesign;
```
Comment (sawantpritam) [inline e2e-test/src/test/java/io/cdap/plugin/datetransform/stepsdesign/package-info.java]: Add javadoc 
and also licence.

Refer- package info file inside utils folder
e2e-test/src/test/java/io/cdap/plugin/utils/package-info.java

File: e2e-test/src/test/resources/pluginDataCyAttributes.properties [hunk 32]
```diff
@@ -0,0 +1,31 @@
...
+pathFileNameOnly=filenameOnly
+readFilesRecursively=recursive
+allowEmptyInput=ignoreNonExistingFolders
+fileSystemProperties=fileSystemProperties
+fileEncoding=fileEncoding
+sourceFieldName=sourceField
+sourceFieldDateFormat=sourceFormat
+targetFieldName=targetField
+targetFieldDateFormat=targetFormat
+sourceinSecondsorMilliseconds?=secondsOrMilliseconds
```
Comment (sawantpritam) [inline e2e-test/src/test/resources/pluginDataCyAttributes.properties]: remove ?

**Review by sawantpritam — CHANGES_REQUESTED** [review]

File: e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature (hunk 5 above)
Comment (Swati10-11) [reply e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature]: @sawantpritam Validation step applied to all scenarios

File: e2e-test/src/test/resources/pluginParameters.properties (hunk 18 above)
Comment (Swati10-11) [reply e2e-test/src/test/resources/pluginParameters.properties]: Changes Done

File: e2e-test/src/test/resources/pluginParameters.properties (hunk 19 above)
Comment (Swati10-11) [reply e2e-test/src/test/resources/pluginParameters.properties]: @sawantpritam Changed Name with Names

File: e2e-test/src/test/resources/pluginParameters.properties (hunk 20 above)
Comment (Swati10-11) [reply e2e-test/src/test/resources/pluginParameters.properties]: @sawantpritam no duplicate

File: e2e-test/src/test/resources/pluginParameters.properties (hunk 21 above)
Comment (Swati10-11) [reply e2e-test/src/test/resources/pluginParameters.properties]: @sawantpritam changes done

File: e2e-test/src/test/resources/testdata/BigQuery/bqDateTransformCreateTableLong.txt (hunk 22 above)
Comment (Swati10-11) [reply e2e-test/src/test/resources/testdata/BigQuery/bqDateTransformCreateTableLong.txt]: @sawantpritam Combine table and deleted extra table

File: e2e-test/src/test/features/datetransformplugin/DateTransformErrorScenarios.feature (hunk 23 above)
Comment (Swati10-11) [reply e2e-test/src/test/features/datetransformplugin/DateTransformErrorScenarios.feature]: @sawantpritam Changes Done

File: e2e-test/src/test/resources/pluginParameters.properties (hunk 24 above)
Comment (Swati10-11) [reply e2e-test/src/test/resources/pluginParameters.properties]: @sawantpritam Removed

File: e2e-test/src/test/features/datetransformplugin/DateTransformErrorScenarios.feature (hunk 25 above)
Comment (Swati10-11) [reply e2e-test/src/test/features/datetransformplugin/DateTransformErrorScenarios.feature]: > Verify DateTransform plugin error for Source and Target field must have same number of fields

Changes done

File: e2e-test/src/test/features/datetransformplugin/DateTransformWithBQ.feature (hunk 26 above)
Comment (Swati10-11) [reply e2e-test/src/test/features/datetransformplugin/DateTransformWithBQ.feature]: > dateTransform.TargetFieldName -> dateTransform.TargetFieldNames

Changes done as  dateTransform.TargetFieldNames

File: e2e-test/src/test/features/datetransformplugin/DateTransformWithBQ.feature (hunk 27 above)
Comment (Swati10-11) [reply e2e-test/src/test/features/datetransformplugin/DateTransformWithBQ.feature]: > remove ? from property name
> 
> Also value property is not added in pluginParameters.properties

Removed ? also added in pluginParameters.properties

File: e2e-test/src/test/java/io/cdap/plugin/datetransform/runners/TestRunner.java (hunk 29 above)
Comment (Swati10-11) [reply e2e-test/src/test/java/io/cdap/plugin/datetransform/runners/TestRunner.java]: > Add multiline comment ->
> 
> `/* TODO: Enable test once issues fixed - https://cdap.atlassian.net/browse/PLUGIN-1224 https://cdap.atlassian.net/browse/PLUGIN-1225 */`

Done Comment line as suggested

File: e2e-test/src/test/java/io/cdap/plugin/datetransform/runners/package-info.java (hunk 14 above)
Comment (Swati10-11) [reply e2e-test/src/test/java/io/cdap/plugin/datetransform/runners/package-info.java]: > Add javadoc and also licence.

Added javadoc and licence

File: e2e-test/src/test/java/io/cdap/plugin/datetransform/stepsdesign/DateTransform.java (hunk 30 above)
Comment (Swati10-11) [reply e2e-test/src/test/java/io/cdap/plugin/datetransform/stepsdesign/DateTransform.java]: > Add one empty line before

Added one empty line

File: e2e-test/src/test/java/io/cdap/plugin/datetransform/stepsdesign/package-info.java (hunk 31 above)
Comment (Swati10-11) [reply e2e-test/src/test/java/io/cdap/plugin/datetransform/stepsdesign/package-info.java]: > Add javadoc and also licence.
> 
> Refer- package info file inside utils folder e2e-test/src/test/java/io/cdap/plugin/utils/package-info.java

Already added there

File: e2e-test/src/test/resources/pluginDataCyAttributes.properties (hunk 32 above)
Comment (Swati10-11) [reply e2e-test/src/test/resources/pluginDataCyAttributes.properties]: > remove ?
Removed

File: e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature (hunk 1 above)
Comment (Swati10-11) [reply e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature]: > rename file -> DateTransformWithBQ.featrure
 Renamed

File: e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature (hunk 2 above)
Comment (Swati10-11) [reply e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature]: > -> Feature: DateTransform - Verification of DateTransform pipeline with BigQuery as source and target

Changes Done

File: e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature (hunk 3 above)
Comment (Swati10-11) [reply e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature]: > -> using DateTransform Plugin for long datatype field

changes done

File: e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature (hunk 4 above)
Comment (Swati10-11) [reply e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature]: > -> using DateTransform Plugin for string datatype field

Changes Done

File: e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature (hunk 5 above)
Comment (Swati10-11) [reply e2e-test/src/test/features/datetransformplugin/datetransformBQ.feature]: > Add format validation step
> 
> Applies to all scenarios

 Validation step applied to all scenarios

File: e2e-test/src/test/features/datetransformplugin/DateTransformErrorScenarios.feature (hunk 25 above)
Comment (sawantpritam) [reply e2e-test/src/test/features/datetransformplugin/DateTransformErrorScenarios.feature]: > > Verify DateTransform plugin error for Source and Target field must have same number of fields
> 
> Changes done

Remove **single**

-> Verify DateTransform plugin error for Source and Target field must have same number of fields

File: e2e-test/src/test/features/datetransformplugin/DateTransformWithBQ.feature (hunk 26 above)
Comment (sawantpritam) [reply e2e-test/src/test/features/datetransformplugin/DateTransformWithBQ.feature]: > > dateTransform.TargetFieldName -> dateTransform.TargetFieldNames
> 
> Changes done as dateTransform.TargetFieldNames

Change not done -> 

dateTransform.TargetFieldName**s**

File: e2e-test/src/test/features/datetransformplugin/datetransformMacros.feature (hunk 28 above)
Comment (sawantpritam) [reply e2e-test/src/test/features/datetransformplugin/datetransformMacros.feature]: > -> with DateTransform plugin properties as macro arguments

change not done ->

with DateTransform **plugin properties** as macro arguments

File: e2e-test/src/test/resources/errorMessage.properties (hunk 16 above)
Comment (sawantpritam) [reply e2e-test/src/test/resources/errorMessage.properties]: > remove all unused - > Required property error messages

Change not done-> remove unused

File: e2e-test/src/test/resources/pluginParameters.properties [hunk 33]
```diff
@@ -0,0 +1,63 @@
...
+readRecursivePathSchema=[{"key":"id","value":"int"},{"key":"name","value":"string"},\
+  {"key":"yearofbirth","value":"int"},{"key":"tablename","value":"string"},{"key":"offset","value":"string"}]
+fileMaxSplitSize=120
+regexPathFilter=.+TEST.csv
+incorrectRegexPathFilter=/tZY
+filePluginOutputFolder=file-plugin-output
+skipHeaderTrue=true
+fileSinkPathSuffix=yyyy-MM-dd-HH-mm-ss
+## FILE-PLUGIN-PROPERTIES-END
+## DateTransform-PLUGIN-PROPERTIES-START
```
Comment (sawantpritam) [inline e2e-test/src/test/resources/pluginParameters.properties]: Add separator empty lines--- removed in latest commit

Also add empty line after line no 3

**Review by sawantpritam — CHANGES_REQUESTED** [review]

File: e2e-test/src/test/resources/pluginParameters.properties (hunk 33 above)
Comment (Swati10-11) [reply e2e-test/src/test/resources/pluginParameters.properties]: > Add separator empty lines--- removed in latest commit
> 
> Also add empty line after line no 3

added empty line

File: e2e-test/src/test/features/datetransformplugin/DateTransformWithBQ.feature (hunk 26 above)
Comment (Swati10-11) [reply e2e-test/src/test/features/datetransformplugin/DateTransformWithBQ.feature]: > > > dateTransform.TargetFieldName -> dateTransform.TargetFieldNames
> > 
> > 
> > Changes done as dateTransform.TargetFieldNames
> 
> Change not done ->
> 
> dateTransform.TargetFieldName**s**

already done changes in last commit


  Then Enter input plugin property: "sourceFieldName" with value: "dateTransform.SourceFieldNames"
    Then Enter input plugin property: "sourceFieldDateFormat" with value: "dateTransform.SourceFieldDateFormat"
    Then Enter input plugin property: "targetFieldName" with value: "dateTransform.TargetFieldNames"
    Then Enter input plugin property: "targetFieldDateFormat" with value: "dateTransform.TargetFieldDateFormat"

File: e2e-test/src/test/features/datetransformplugin/DateTransformErrorScenarios.feature (hunk 25 above)
Comment (Swati10-11) [reply e2e-test/src/test/features/datetransformplugin/DateTransformErrorScenarios.feature]: > > > Verify DateTransform plugin error for Source and Target field must have same number of fields
> > 
> > 
> > Changes done
> 
> Remove **single**
> 
> -> Verify DateTransform plugin error for Source and Target field must have same number of fields

Changes done

 Verify DateTransform plugin error for Source and Target field must have same number of fields

**Review by sawantpritam — APPROVED** [review]

**Comment by sawantpritam** [comment]
Squash the commits.
